### PR TITLE
feat: Add a asset loading helper to abstract away the asset loading internals

### PR DESCRIPTION
**Pull request by Rhuagh**
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/1090)

### CODE REVIEW

**Comment by torkleyy** [comment]
I'd like to review before this grts merged. Maybe we can get rid of the
helper suffix somehow?

**Comment by Rhuagh** [comment]
@torkleyy suggestions are welcome, i'm also not happy with the name but it's the best i could come up with

File: amethyst_assets/src/helper.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
+use std::borrow::Borrow;
+use std::hash::Hash;
+
+use amethyst_core::specs::{Read, ReadExpect};
+
+use {Asset, AssetStorage, Format, Handle, Loader, Progress};
+
+/// Helper type for loading assets
+#[derive(SystemData)]
+pub struct AssetLoadHelper<'a, A>
```
[the remark this answers is not in the document]
Comment (azriel91) [reply amethyst_assets/src/helper.rs]: what about `SystemDataLoader`? I know the name's longer, but it's more specific than `Helper`

[the remark this answers is not in the document]
Comment (torkleyy) [reply amethyst_assets/src/helper.rs]: hmm, think `SystemData` should be the suffix in this case.

**Comment by Rhuagh** [comment]
Not sure that names better, but there you go.

**Comment by Rhuagh** [comment]
bors r=torkleyy,jojolepro

**Pull request merged**